Game/MyProject/Game/Gameplay/InfoManager.cpp

```diff
@@ -0,0 +1,99 @@
+#include "InfoManager.hpp"
+
+InfoManager::InfoManager(SceneNode* root) : root(root) {
+    
+    //get list of all cars
+    std::vector<std::shared_ptr<SceneNode>>cars;
+    root->findAllNodesContainingName("Car", cars);
+    for(auto& i : cars)
+        carsList.push_back(i.get());
```

Comment:
no need to iterate cars twice, you can call `setLap`, `setCheckpoint` and `setScore` here.